internal/api/v1/services/instance.go

```diff
@@ -103,77 +104,120 @@ func (s *Instance) GetInstance(ctx context.Context, ownerID, id uint) (*models.I
 	return s.repo.Get(ctx, ownerID, id)
 }
 
+// updateInstanceVolumes updates the volumes and volume details for an instance
+func (s *Instance) updateInstanceVolumes(
+	ctx context.Context,
+	instanceName string,
+	volumes []string,
+	volumeDetails []types.VolumeDetails,
+) error {
+	// Get the instance first to get its ownerID
+	instance, err := s.repo.GetByName(ctx, instanceName)
+	if err != nil {
+		return fmt.Errorf("failed to get instance %s: %w", instanceName, err)
+	}
+
+	// Convert volume details to database model
+	var dbVolumeDetails []models.VolumeDetail
+	for _, vd := range volumeDetails {
+		dbVolumeDetails = append(dbVolumeDetails, models.VolumeDetail{
+			ID:         vd.ID,
+			Name:       vd.Name,
+			SizeGB:     vd.SizeGB,
+			MountPoint: vd.MountPoint,
+		})
+	}
```

Comment:
Why not to use one type for all. like defining the db model type and use it here also so no need to convert?